include/envoy/thread_local/thread_local.h

```diff
@@ -16,49 +16,29 @@ namespace ThreadLocal {
 class ThreadLocalObject {
 public:
   virtual ~ThreadLocalObject() {}
-
-  /**
-   * The owning thread is about to exit. Cleanup any variables that may reference other thread
-   * locals.
-   */
-  virtual void shutdown() PURE;
 };
 
 typedef std::shared_ptr<ThreadLocalObject> ThreadLocalObjectSharedPtr;
```

Comment:
Though I agree that would be awesome, in this case, we actually can't do that universally, since there are cases where we shared read-only data across threads globally.